homeassistant/components/device_tracker/meraki.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import json
+
+import voluptuous as vol
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import (HTTP_BAD_REQUEST, HTTP_UNPROCESSABLE_ENTITY)
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.components.device_tracker import (
+    PLATFORM_SCHEMA, SOURCE_TYPE_ROUTER)
+
+CONF_VALIDATOR = 'validator'
+CONF_SECRET = 'secret'
+DEPENDENCIES = ['http']
+URL = '/api/meraki'
+VERSION = '2.0'
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_VALIDATOR): cv.string,
+    vol.Required(CONF_SECRET): cv.string
+})
+
+
+@asyncio.coroutine
+def async_setup_scanner(hass, config, async_see, discovery_info=None):
+    """Set up an endpoint for the Meraki tracker."""
+    hass.http.register_view(
+        MerakiView(config, async_see))
+
+    return True
+
+
+class MerakiView(HomeAssistantView):
+    """View to handle Meraki requests."""
+
+    url = URL
+    name = 'api:meraki'
+
+    def __init__(self, config, async_see):
+        """Initialize Meraki URL endpoints."""
+        self.async_see = async_see
+        self.validator = config[CONF_VALIDATOR]
+        self.secret = config[CONF_SECRET]
+
+    @asyncio.coroutine
+    def get(self, request):
+        """Meraki message received as GET."""
+        return self.validator
+
+    @asyncio.coroutine
+    def post(self, request):
+        """Meraki CMX message received."""
+        try:
+            data = yield from request.json()
+        except ValueError:
+            return self.json_message('Invalid JSON', HTTP_BAD_REQUEST)
+        _LOGGER.debug("Meraki Data from Post: %s", json.dumps(data))
+        if not data.get('secret', False):
+            _LOGGER.error("secret invalid")
+            return self.json_message('No secret', HTTP_UNPROCESSABLE_ENTITY)
+        if data['secret'] != self.secret:
+            _LOGGER.error("Invalid Secret received from Meraki")
+            return self.json_message('Invalid secret',
+                                     HTTP_UNPROCESSABLE_ENTITY)
+        elif data['version'] != VERSION:
+            _LOGGER.error("Invalid API version: %s", data['version'])
+            return self.json_message('Invalid version',
+                                     HTTP_UNPROCESSABLE_ENTITY)
+        else:
+            _LOGGER.debug('Valid Secret')
+            if data['type'] == "DevicesSeen":
+                _LOGGER.debug("WiFi Devices Seen")
+            elif data['type'] == "BluetoothDevicesSeen":
+                _LOGGER.debug("Bluetooth Devices Seen")
+            else:
+                _LOGGER.error("Unknown Device %s", type)
+                return self.json_message('invalid device type',
+                                         HTTP_UNPROCESSABLE_ENTITY)
+        if len(data["data"]["observations"]) == 0:
+            _LOGGER.debug("No observations found")
+            return
+        yield from self._handle(request.app['hass'], data)
```

Comment:
When you change `_handle` to a callback function instead of a coroutine (see comment below), change here to:
```py
self._handle(request.app['hass'], data)
```